evaluation.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+from hourglass import HourglassNet
+from data_generator import DataGenerator
+from constants import *
+from util import *
+
+import time
+from datetime import timedelta
+import matplotlib.pyplot as plt
+import numpy as np
+from PIL import Image
+import glob
+import cv2
+
+def predict(hourglass, model_json, model_weights, subset):
+    hgnet = HourglassNet(num_classes=NUM_COCO_KEYPOINTS, num_stacks=hourglass, num_channels=NUM_CHANNELS, inres=INPUT_DIM,
+                            outres=OUTPUT_DIM)
+    hgnet._load_model(model_json, model_weights)
+
+    val_df = hgnet.load_and_filter_annotations(DEFAULT_TRAIN_ANNOT_PATH, DEFAULT_VAL_ANNOT_PATH, subset)
+
+    generator = DataGenerator(
+        df=val_df,
+        base_dir=DEFAULT_TRAIN_IMG_PATH,
+        input_dim=INPUT_DIM,
+        output_dim=OUTPUT_DIM,
+        num_hg_blocks=DEFAULT_NUM_HG,
+        shuffle=False,  
+        batch_size=1,
+        online_fetch=False)
+
+    # Select image to predict heatmaps
+    X_batch, y_stacked = generator[168]
+    X = X_batch[0] # take first example of batch
+
+    X = X.reshape(1, 256, 256, 3) # predict needs shape (1, 256, 256, 3)
+    output = hgnet.model.predict(X)
+    output = np.array(output) # output shape is (hourglass_num, 1, 64, 64, 17)
+
+    # Save output heatmaps
+    for i in range(NUM_COCO_KEYPOINTS):
+        plt.figure(i)
+        plt.imsave(f'image{i}.png', output[3, 0, :, :, i])
+
+    # Overlay heatmaps
+    # https://automaticaddison.com/how-to-blend-multiple-images-using-opencv/
+    # Import all image files with image*.png
+    files = glob.glob ("image*.png")
+    image_data = []
+    for my_file in files:
+        this_image = cv2.imread(my_file, 1)
+        image_data.append(this_image)
+    
+    # Calculate blended image
+    dst = image_data[0]
+    for i in range(len(image_data)):
+        if i == 0:
+            pass
+        else:
+            alpha = 1.0/(i + 1)
+            beta = 1.0 - alpha
+            dst = cv2.addWeighted(image_data[i], alpha, dst, beta, 0.0)
+ 
+    # Save blended heatmap image
+    cv2.imwrite('result.png', dst)
```

Comment:
More of quality of life comment, but were you able to get any success with matplot working in Collab so that you can view image overlays in a notebook instead of needing to write to disk?

Reply:
So I'm currently able to view all plots in Collab/when saved to Drive but not locally, now I'm also running into issues viewing heatmaps when they are saved to local disk, I'm not sure if everyone else is having the same issues.

Reply:
If it ends up being easier, you can always try moving the visualization logic to a notebook. That's what I did for `coco-explorer`

Reply:
If you've used the `# %%` short form for a normal `.py` file, you can right click on the file in VS code and convert to a python notebook too